presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkRemoteSourceFactory.java

```diff
@@ -18,52 +18,89 @@
 import com.facebook.presto.common.block.SortOrder;
 import com.facebook.presto.common.type.Type;
 import com.facebook.presto.operator.SourceOperatorFactory;
+import com.facebook.presto.spark.classloader_interface.PrestoSparkDiskPage;
 import com.facebook.presto.spark.classloader_interface.PrestoSparkSerializedPage;
 import com.facebook.presto.spark.classloader_interface.PrestoSparkShuffleStats;
+import com.facebook.presto.spark.classloader_interface.PrestoSparkTaskOutput;
 import com.facebook.presto.spark.execution.PrestoSparkRemoteSourceOperator.SparkRemoteSourceOperatorFactory;
 import com.facebook.presto.spi.plan.PlanNodeId;
+import com.facebook.presto.spi.storage.TempDataOperationContext;
+import com.facebook.presto.spi.storage.TempStorage;
 import com.facebook.presto.sql.planner.RemoteSourceFactory;
 import com.google.common.collect.ImmutableMap;
 import org.apache.spark.util.CollectionAccumulator;
 
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
+import static com.facebook.presto.SystemSessionProperties.isDistributeBroadcastTableUsingDisk;
 import static com.facebook.presto.spark.util.PrestoSparkUtils.createPagesSerde;
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
-public class PrestoSparkRemoteSourceFactory
+public class PrestoSparkRemoteSourceFactory<T extends PrestoSparkTaskOutput>
```

Comment:
Generic isn't necessary here. `Map<PlanNodeId, List<List<T>>> broadcastInputsMap` can be defined as `Map<PlanNodeId, List<?>> broadcastInputsMap`